### DISCUSSION TITLE
this sql expression is error  but i don't konw where error

### DISCUSSION THREAD

**Discussion by q1165600785** [sqlalchemy/sqlalchemy#9579] [Usage Questions]
def qurer_data_antmod_name(self,conn,antmod_name):
        subq = (select(antmod_one_one.c.id,antmod_one_one.c.model_name,
                       antmod_one_one.c.antmod_name,antmod_one_one.c.rad,
                       antmod_one_one.c.dh,antmod_one_one.c.L1_E,
                       antmod_one_one.c.L1_N,antmod_one_one.c.L1_U,
                       antmod_one_one.c.L2_E, antmod_one_one.c.L2_N,
                       antmod_one_one.c.L2_U,antmod_one_one.c.remark
                      ).where(antmod_one_one.c.antmod_name.like("%" + antmod_name + "%")).subquery())
        print(subq)
        stmt=select(subq, func.concat_ws(func.group_concat(antmod_picture.c.picture_dir),
                    func.group_concat(model_picture.c.picture_dir)).label("picture_dir")
                    ).join_from(subq,model_picture,subq.c.model_name==model_picture.c.model).join_from(
        subq, antmod_picture, subq.c.antmod_name == antmod_picture.c.antmod).group_by(subq.c.id)
        print(stmt)
        qry = conn.execute(statement,execution_options={"prebuffer_rows": True})
        return qry

Traceback (most recent call last):
  File "E:\python\anconda\envs\query_antmod_env\lib\site-packages\sqlalchemy\sql\base.py", line 1565, in __getattr__
    return self._index[key][1]
KeyError: 'model'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "I:/gamit/query_antmod_v3/data_base_z/database_connection/database_connection.py", line 200, in <module>
    result=get_opr.query_data_model_name(conn,"zz")
  File "I:/gamit/query_antmod_v3/data_base_z/database_connection/database_connection.py", line 59, in query_data_model_name
    ).join_from(subq, model_picture, subq.c.model_name == model_picture.c.model).join_from(
  File "E:\python\anconda\envs\query_antmod_env\lib\site-packages\sqlalchemy\sql\base.py", line 1567, in __getattr__
    raise AttributeError(key) from err
AttributeError: model

**Comment by q1165600785** [comment]
antmod_one_one=Table("antmod_one_one",metadata_obj,
    Column("id",BIGINT, primary_key=True),
    Column("model_name",VARCHAR(50)),
    Column("antmod_name",VARCHAR(50)),
    Column("dh",FLOAT(5,2)),
    Column("rad",FLOAT(5,2)),
    Column("L1_N",FLOAT(5,2)),
    Column("L1_E",FLOAT(5,2)),
    Column("L1_U",FLOAT(5,2)),
    Column("L2_N",FLOAT(5,2)),
    Column("L2_E",FLOAT(5,2)),
    Column("L2_U",FLOAT(5,2)),
    Column("add_time",DATETIME,default=datetime.datetime.now,onupdate=datetime.datetime.now),
    Column("source",TINYINT,default=1,onupdate=2),
    Column("remark",TEXT),
    )
model_picture=Table("model_picture",metadata_obj,
    Column("id",BIGINT, primary_key=True),
    Column("antmod",VARCHAR(50)),
    Column("picture_dir",VARCHAR(150)),
    Column("add_time",DATETIME,default=datetime.datetime.now),
    )

antmod_picture=Table("antmod_picture",metadata_obj,
    Column("id",BIGINT, primary_key=True),
    Column("model",VARCHAR(50)),
    Column("picture_dir",VARCHAR(150)),
    Column("add_time",DATETIME,default=datetime.datetime.now)
    )

**Reply by q1165600785** [reply]
if __name__ == '__main__':
    engine = create_engine(
        URL.create("mysql+pymysql",
                   username="root",
                   password="zhangyujie",
                   host="localhost",
                   port=3306,
                   database="antmod_database"), echo=True)
    with engine.connect() as conn:
        print(insert_dict_list)
        get_opr =model_table_opr()
        result=get_opr.query_data_model_name(conn,"zz")
        conn.commit()
        print(result)

**Comment by CaselIT** [comment]
Hi

You are doing `model_picture.c.model` but that table has no `model` column
